src/journal/src/Journal/Internal/ByteBufferPtr.hs

```diff
@@ -117,30 +117,32 @@ remaining bb = do
 
 boundCheck :: HasCallStack => ByteBuffer -> Int -> IO ()
 boundCheck bb ix = do
+  invariant bb
   -- XXX: parametrise on build flag and only do these checks if enabled?
   Slice slice <- readIORef (bbSlice bb)
-  if ix - slice < fromIntegral (getCapacity bb) &&
-     0 <= ix + slice
+  Limit limit <- readIORef (bbLimit bb)
+  if 0 <= ix + slice &&
+     ix - slice < limit - slice
```

Comment:
```suggestion
  Limit limit <- readIORef (bbLimit bb)
  if 0 <= ix &&
     ix < limit
```
Should we really check against the `slice` here? Shouldn't we check that `ix` is in the range `[0, limit)`? Maybe we want another check that `ix + slice` is in `[0, capacity)` but not entirely sure about that one (since we don't keep the old original capacity).